SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/backend/ToJitInnerVisitor.java

```diff
@@ -0,0 +1,706 @@
+
+        if (sourceId.hasNull()) {
+            ObjectNode isNull = ev.getNullObject().putObject("CopyVal");
+            isNull.put("value", sourceId.isNullId);
+            isNull.put("value_ty", "Bool");
+        }
+        return false;
+    }
+
+    static final Map<String, String> opNames = new HashMap<>();
+
+    static {
+        // https://github.com/vmware/database-stream-processor/blob/dataflow-jit/crates/dataflow-jit/src/ir/expr.rs, the BinaryOpKind enum
+        opNames.put("+", "Add");
+        opNames.put("-", "Sub");
+        opNames.put("*", "Mul");
+        opNames.put("/", "Div");
+        opNames.put("==", "Eq");
+        opNames.put("!=", "Neq");
+        opNames.put("<", "LessThan");
+        opNames.put(">", "GreaterThan");
+        opNames.put("<=", "LessThanOrEqual");
+        opNames.put(">=", "GreaterThanOrEqual");
+        opNames.put("&", "And");
+        opNames.put("&&", "And");
+        opNames.put("|", "Or");
+        opNames.put("||", "Or");
+        opNames.put("^", "Xor");
+    }
+
+    @Override
+    public boolean preorder(DBSPBinaryExpression expression) {
+        // "BinOp": {
+        //   "lhs": 4,
+        //   "rhs": 5,
+        //   "operand_ty": "I64",
+        //   "kind": "GreaterThan"
+        // }
+        ExpressionIds leftId = this.accept(expression.left);
+        ExpressionIds rightId = this.accept(expression.right);
+        ExpressionIds cf = this.constantBool(false);
+        ExpressionRepresentation er = this.insertInstruction(expression);
+
+        // The following are not used on all execution paths
+        int leftNullId;
+        if (leftId.hasNull())
+            leftNullId = leftId.isNullId;
+        else
+            // Not nullable: use false.
+            leftNullId = cf.id;
+        int rightNullId;
+        if (rightId.hasNull())
+            rightNullId = rightId.isNullId;
+        else
+            rightNullId = cf.id;
+
+        if (expression.operation.equals("||") &&
+                expression.getNonVoidType().is(DBSPTypeBool.class)) {
+           throw new Unimplemented(expression);
+        } else if (expression.operation.equals("&&")) {
+            throw new Unimplemented(expression);
+        } else {
+            ObjectNode binOp = er.instruction.putObject("BinOp");
+            binOp.put("lhs", leftId.id);
+            binOp.put("rhs", rightId.id);
+            binOp.put("kind", Utilities.getExists(opNames, expression.operation));
+            binOp.put("operand_ty", baseTypeName(expression.left));
+            if (er.isNullInstruction != null) {
+                // The result is null if either operand is null.
+                binOp = er.isNullInstruction.putObject("BinOp");
+                binOp.put("lhs", leftNullId);
+                binOp.put("rhs", rightNullId);
+                binOp.put("kind", Utilities.getExists(opNames, "||"));
+                binOp.put("operand_ty", "Bool");
+            }
+        }
```

Comment:
```suggestion
        ObjectNode binOp = er.instruction.putObject("BinOp");
        binOp.put("lhs", leftId.id);
        binOp.put("rhs", rightId.id);
        binOp.put("kind", Utilities.getExists(opNames, expression.operation));
        binOp.put("operand_ty", baseTypeName(expression.left));
        if (er.isNullInstruction != null) {
            // The result is null if either operand is null.
            binOp = er.isNullInstruction.putObject("BinOp");
            binOp.put("lhs", leftNullId);
            binOp.put("rhs", rightNullId);
            binOp.put("kind", Utilities.getExists(opNames, "||"));
            binOp.put("operand_ty", "Bool");
        }
```

Reply:
Unfortunately that's not how SQL works:
https://www.postgresql.org/docs/current/functions-logical.html

Reply:
What about it is different? Logical and/or are logical and/or, they're identical from what I can see

Reply:
TRUE || NULL = TRUE instead of NULL

Reply:
Can't you just correct the nullish setting stuff then?